### PR TITLE
chore: update for Python 3.7 syntax

### PR DESCRIPTION
**Pull request by henryiii**
This makes the following changes:

- chore: update style settings - Modifying checks should come first, and adding isort
- chore: fix line endings - Windows line endings seem to have slipped in for 6-8 files or so, check your [git settings](https://docs.github.com/en/github/getting-started-with-github/configuring-git-to-handle-line-endings)
- style: run pre-commit (sort includes) - Automated run of the changes in the first commit
- chore: use from future import annotations - Used isort to inject `from __future__ import annotations`
- style: run pre-commit (use 3.7+ annotations) - Let pyupgrade clean up the quoted annotations
- "chore: cleanup imports, renable E402" - Turns this check on, drops unneeded imports or adds them to `__all__`

### CODE REVIEW

File: src/mplhep/plot.py [hunk 1]
```diff
@@ -1,915 +1,914 @@
...
+        _labels = _labels[::-1]
+        _chunked_kwargs = _chunked_kwargs[::-1]
+
+    #################################
+    # Split and reshape for iteration
+    h = h.reshape(len(hists), -1)
+
+    ##########
+    # Plotting
+    return_artists: list[StairsArtists | ErrorBarArtists] = []
```
Comment (henryiii) [inline src/mplhep/plot.py:325]: This is an example of the 3.7+ cleanup I was after.

File: src/mplhep/plot.py [hunk 2]
```diff
@@ -1,915 +1,914 @@
...
+        else:
+            for i in range(len(_chunked_kwargs)):
+                _chunked_kwargs[i][kwarg] = kwargs[kwarg]
+
+    _bin_widths = np.diff(final_bins)
+    _bin_centers = final_bins[1:] - _bin_widths / float(2)
+
+    ############################
+    # yerr calculation
+    _yerr: np.ndarray | None
```
Comment (henryiii) [inline src/mplhep/plot.py:223]: And this.

**Comment by henryiii** [comment]
Be sure to "temporarily ignore whitespace" on the GitHub diff to see what I'm referring to above! Otherwise, it confuses the line endings changes.

**Comment by andrzejnovak** [comment]
Thanks @henryiii, sorry for the delay, for some reason I did not get a notification for this.

**Pull request merged**